### DISCUSSION TITLE
support Console Do Not Track env var: DO_NOT_TRACK

### DISCUSSION THREAD

**Discussion by derhuerst** [npm/feedback#481] [CLI] [answered]
Hey! 👋

I suggest that the npm CLI respects the Console Do Not Track (DNT) environment variable `DO_NOT_TRACK`.

Many `npm` users probably don't care about periodic update checks, and this change wouldn't. But those who do should have a reasonable way to deny tracking/telemetry; This is in line with recent efforts by Apple (among others) to make the tracking consent more informed.

> Console Do Not Track (DNT)

> This is a proposal for a single, standard environment variable that plainly and unambiguously expresses LACK OF CONSENT > by a user of that software to any of the following:
>
> - ad tracking
> - usage reporting, anonymous or not
> - automatic update phone-home
> - crash reporting
> - non-essential-to-functionality requests of any kind to the creator of the software or other tracking services

> ```
> export DO_NOT_TRACK=1
> ```

Of course, most network requests sent by the `npm` CLI *are* essential to its functionality, but at least the [`update-notifier` config](https://docs.npmjs.com/cli/v7/using-npm/config#update-notifier) could be set to `false`.

**Comment by MylesBorins** [comment, marked as answer]
@derhuerst this seems like a great place to start with an RFC over at https://github.com/npm/rfcs about the behavior change... likely start with an issue because it may not require a full RFC if there is general consensus about the behavior change
